docs/01-app/03-api-reference/05-config/01-next-config-js/watchOptions.mdx

```diff
@@ -0,0 +1,88 @@
+---
+title: watchOptions
+description: Enable file watch polling for Turbopack
+---
+
+Enable file watch polling for Turbopack.
+
+## Usage
+
+The `watchOptions` option allows you to customize how Turbopack watches files during development.
+
+```ts filename="next.config.ts" switcher
+import type { NextConfig } from 'next'
+
+const nextConfig: NextConfig = {
+  watchOptions: {
+    pollIntervalMs: 1000,
+  },
+}
+
+export default nextConfig
+```
+
+```js filename="next.config.js" switcher
+/** @type {import('next').NextConfig} */
+const nextConfig = {
+  watchOptions: {
+    pollIntervalMs: 1000,
+  },
+}
+
+module.exports = nextConfig
+```
+
+## Config options
+
+- `pollIntervalMs`: How often Turbopack polls for file changes.
+
+## When to use
+
+Enable this option if you're running into issues where file changes are not being detected during development.
+
+This can happen when working inside a VM, Docker container, or over a networked volume.
+
+### Webpack equivalent
+
+If you're using Webpack, you can achieve similar behavior by setting the `poll` option:
+
+```ts filename="next.config.ts" switcher
+import type { NextConfig } from 'next'
+
+const nextConfig: NextConfig = {
+  webpack(config) {
+    config.watchOptions = {
+      ...config.watchOptions,
+      poll: 1000,
+    }
+    return config
+  },
+}
+
+export default nextConfig
+```
+
+```js filename="next.config.js" switcher
+/** @type {import('next').NextConfig} */
+const nextConfig = {
+  webpack(config) {
+    config.watchOptions = {
+      ...config.watchOptions,
+      poll: 1000,
```

Comment:
For what it's worth, you can also configure this via an env variable.

E.g. in `package.json`

```json
"scripts": {
  "dev": "WATCHPACK_POLLING=1000 next dev"
}
```